env0/validators.go

```diff
@@ -129,3 +117,29 @@ func ValidateTtl(i interface{}, path cty.Path) diag.Diagnostics {
 
 	return nil
 }
+
+func NewRoleValidator(supportedBuiltInRoles []string) schema.SchemaValidateDiagFunc {
```

Comment:
I don't thing `New` as a prefix in the name is a good choice.
New until when? 🙃

can it be just `RoleValidator`?

Reply:
It creates a call back function that acts as the validator (based on the given input).
That's why a validator is used.

Reply:
I didnt get your response, I am talking about the function name

Reply:
Notice it returns a new function.
In other words, it creates something.
Some validators do not create anything and are without the "new" some do and have the "new".

Reply:
I'll merge - if it's still an issue.
We can discuss more - the issue will remain open anyways, as I plan another final PR.